packages/wallet/src/domain/commitments/index.ts

```diff
@@ -45,6 +46,46 @@ export function constructConclude(commitment: Commitment): Commitment {
   return { ...incrementTurnNum(commitment), commitmentType: CommitmentType.Conclude };
 }
 
+export function nextLedgerUpdateCommitment(
+  commitment: Commitment,
+): Commitment | 'NotAnUpdateCommitment' {
+  const numParticipants = commitment.channel.participants.length;
+  const turnNum = commitment.turnNum + 1;
+  if (commitment.commitmentType !== CommitmentType.App || turnNum <= 2 * numParticipants) {
+    return 'NotAnUpdateCommitment';
+  }
+
+  let allocation;
+  let destination;
+  let updatedAppAttributes;
+
+  // This is the final commitment so we must reset to 0 and update allocation/destination
+  if (turnNum === 3 * numParticipants - 1) {
```

Comment:
Should this not happen on, e.g. `turnNum === 3` ?

Assuming `numParticipants = 2`, the RHS of this condition evaluates to 5.